commands/delete-repository.ts

```diff
@@ -0,0 +1,167 @@
+        return;
+    }
+    let registry: Registry;
+    let subscription: SubscriptionModels.Subscription;
+    let repoName: string;
+    let username: string;
+    let password: string;
+    if (!context) {
+        //first get desired registry
+        let registries = await AzureCredentialsManager.getInstance().getRegistries();
+        let reg: string[] = [];
+        for (let i = 0; i < registries.length; i++) {
+            reg.push(registries[i].name);
+        }
+        let desired = await vscode.window.showQuickPick(reg, { 'canPickMany': false, 'placeHolder': 'Choose the Registry from which your desired repository exists' });
+        if (desired === undefined) return;
+        registry = registries.find(reg => { return desired === reg.name });
+
+        //get the subscription object by using the id found on the registry id
+        let subscriptionId = registry.id.slice('/subscriptions/'.length, registry.id.search('/resourceGroups/'));
+        const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+        subscription = subs.find(function (sub): boolean {
+            return sub.subscriptionId === subscriptionId;
+        });
+
+        //get the desired repository to delete
+        const myRepos: azureUtils.Repository[] = await azureUtils.getAzureRepositories(registry);
+        let rep: string[] = [];
+        for (let j = 0; j < myRepos.length; j++) {
+            rep.push(myRepos[j].name);
+        }
+        let desiredRepo = await vscode.window.showQuickPick(rep, { 'canPickMany': false, 'placeHolder': 'Choose the repository you want to delete' });
+        if (desiredRepo === undefined) return;
+        let repository = myRepos.find(rep => { return desiredRepo === rep.name });
+        if (repository === undefined) return;
+        repoName = repository.name;
+    }
+
+    //ensure user truly wants to delete registry
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: true,
+        placeHolder: 'No',
+        value: 'No',
+        prompt: 'Are you sure you want to delete this repository and its associated images? Enter Yes or No: '
+    };
+    let answer = await vscode.window.showInputBox(opt);
+    if (answer !== 'Yes') return;
+
+    // generate credentials before requesting a delete.
+    if (context) {
+        username = context.userName;
+        password = context.password;
+        repoName = context.label;
+        subscription = context.subscription;
+        registry = context.registry;
+    }
+    else { //this is separated from !context above so it only calls loginCredentials once user has assured they want to delete the repository
+        let creds = await loginCredentials(subscription, registry);
+        username = creds.username;
+        password = creds.password;
+    }
+    let path = `/v2/_acr/${repoName}/repository`;
+    let r = await request_data_from_registry('delete', registry.loginServer, path, username, password);
```

Comment:
Do you need to store this in a variable?

Reply:
we do not!